main/src/library/BPlusTree.flix

```diff
@@ -311,6 +311,48 @@ mod BPlusTree {
                 }
         }
 
+    ///
+    /// Returns `true` if and only if for all `k => v` in `t`, `f(k, v) == true`.
+    ///
+    /// Not thread-safe.
+    ///
+    pub def forAll(f: k -> v -> Bool \ ef, t: BPlusTree[k, v, r]): Bool \ ef + r = 
```

Comment:
If you implement forAll (and exist) directly then you can take advantage of short circuiting, right? E.g. you dont have to visit the entire tree if you find false.